### PR TITLE
Sanitize strings when generating VM name

### LINKED ISSUE #265: ibmcloud: sanitise nodeName if it contains none valid characters
The rules for naming vsi instances in ibmcloud are:

1. must contain lowercase alphanumeric characters and hyphens only (without spaces).
2. must have a lowercase letter for the first character.

Currently if you have a nodeName which doesn't match these rules. i.e. it is set to an ip address `10.0.0.1` it will fail to create a instance with the name `10.0.0.1-default-nginx-9348e9fa` for example.

Proposed solution is to replace all none valid characters with `-` resulting in  `10-0-0-1-default-nginx-9348e9fa`. We must then ensure it is prefixed with an lowercase letter. This could be `ip-` or `node-` or `anything-`.

### PR DESCRIPTION
**Pull request by yoheiueda**
This PR sanitizes input string values, and then generates a VM name.

This function generates a VM instance name from sanitized values of node name, pod namespace, pod name, and sandbox ID.

Fixes #265

### CODE REVIEW

**Review by littlejawa — DISMISSED** [review]

File: pkg/util/hvutil/hvutil.go [hunk 1]
```diff
@@ -23,3 +24,33 @@ func GetPodNamespace(annotations map[string]string) string {
...
+		if !(('a' <= c && c <= 'z') || ('0' <= c && c <= '9') || c == '-') {
+			c = '-'
+		}
+		output += string(c)
+	}
+
+	return output
+}
+
+func GetInstanceName(nodeName, podNamespace, podName, sandboxID string) string {
```
Comment (bpradipt) [inline pkg/util/hvutil/hvutil.go:42]: nit: technically this is creating the name of the instance, so should this method be instead called `MakeInstanceName` or `CreateInstanceName` ?

File: pkg/util/hvutil/hvutil.go [hunk 2]
```diff
@@ -23,3 +24,33 @@ func GetPodNamespace(annotations map[string]string) string {
...
+
+	nodeName = sanitize(nodeName)
+	podNamespace = sanitize(podNamespace)
+	podName = sanitize(podName)
+	sandboxID = sanitize(sandboxID)
+
+	vmName := fmt.Sprintf("%s-%s-%s-%.8s", nodeName, podNamespace, podName, sandboxID)
+
+	if c := vmName[0]; !('a' <= c && c <= 'z') {
+		vmName = "ip-" + vmName
```
Comment (bpradipt) [inline pkg/util/hvutil/hvutil.go:52]: nit: how about using a prefix unique to caa (eg. `podvm` or `caa-vm-`) to make it easier for querying ?
Also should this method also restrict the name to a specified length ?

Comment (yoheiueda) [reply pkg/util/hvutil/hvutil.go:52]: OK. I'll use prefix 'podvm`.

File: pkg/util/hvutil/hvutil.go (hunk 1 above)
Comment (yoheiueda) [reply pkg/util/hvutil/hvutil.go:42]: OK. I'll rename to `CreateInstanceName`

**Review by bpradipt — APPROVED** [review]
/lgtm

**Review by littlejawa — APPROVED** [review]
lgtm
Thanks!

**Pull request merged**